File: C#/AnaliseDeSentimento/AnaliseDeSentimento/Arquivo/GerarArquivo.cs
using AnaliseDeSentimento.Entidades;
using AnaliseDeSentimento.Constantes;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;

namespace AnaliseDeSentimento.Arquivo
{
    public class GerarArquivo
    {
        public string NomeArquivo { get; init; }
        public string Endereco { get; init; }
        public EnumClassificacao? TipoAvaliacao { get; init; }

        public GerarArquivo(EnumClassificacao tipoAvaliacao)
        {
            TipoAvaliacao = tipoAvaliacao;
            NomeArquivo = SelecionarNomeArquivo();
            Endereco = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, NomeArquivo);
        }

        public GerarArquivo()
        {
            NomeArquivo = SelecionarNomeArquivo();
            Endereco = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, NomeArquivo);
        }

        string SelecionarNomeArquivo()
            => TipoAvaliacao switch
            {
                EnumClassificacao.Positiva => NomeArquivos.ClassificadosPositivos,
                EnumClassificacao.Neutra => NomeArquivos.ClassificadosNeutros,
                EnumClassificacao.Negativa => NomeArquivos.ClassificadosNegativos,
                _ => NomeArquivos.Output
            };

        public void EscreverRestaurantes(List<Restaurante> listaRestaurantes)
        {
            if (TipoAvaliacao is null)
                return;

            RemoverArquivoSeExistir();
            using var file = File.CreateText(Endereco);

            listaRestaurantes
                .Where(restaurante => restaurante.Classificacao == TipoAvaliacao)
                .ToList()
                .ForEach(restaurante =>
                {
                    file.WriteLine(restaurante);
                });
        }

        public void EscreverSaida(Classificacao classificacao)
        {
            RemoverArquivoSeExistir();
            using var file = File.CreateText(Endereco);

            file.WriteLine($"Restaurantes classificados corretamente: {classificacao.ClassificadosCorretamente.Count}");
            foreach (var restaurante in classificacao.ClassificadosCorretamente)
                file.WriteLine(restaurante.Nome);

            file.WriteLine();
            file.WriteLine($"Restaurantes classificados incorretamente: {classificacao.ClassificadosIncorretamente.Count}");
            foreach (var restaurante in classificacao.ClassificadosIncorretamente)
                file.WriteLine(restaurante.Nome);

            file.WriteLine();
            file.WriteLine($"Acurácia do modelo: {classificacao.Acuracia}");
        }

        void RemoverArquivoSeExistir()
        {
            if (File.Exists(Endereco))
                File.Delete(Endereco);
        }
    }
}

File: Python/Arquivo/GerarArquivo.py
import os
from Constantes.EnumClassificacao import EnumClassificacao
from Constantes.NomeArquivos import NomeArquivos
from Entidades.Classificacao import Classificacao


class GerarArquivo:
  def __init__(self, tipoAvaliacao):
    self.TipoAvaliacao = tipoAvaliacao
    self.NomeArquivo = self.SelecionarNomeArquivo()
    self.Endereco = os.path.dirname(os.path.realpath(__file__)) + "/" + self.NomeArquivo

  def SelecionarNomeArquivo(self):
    if self.TipoAvaliacao is EnumClassificacao.Positiva:
      return NomeArquivos.ClassificadosPositivos

    if self.TipoAvaliacao is EnumClassificacao.Neutra:
      return NomeArquivos.ClassificadosNeutros

    if self.TipoAvaliacao is EnumClassificacao.Negativa:
      return NomeArquivos.ClassificadosNegativos

    return NomeArquivos.Output
    
  def EscreverRestaurantes(self, listaRestaurantes):
    if self.TipoAvaliacao is None:
      return

    arquivo = open(self.Endereco, "w")
    for restaurante in listaRestaurantes:
      if restaurante.Classificacao is self.TipoAvaliacao:
        arquivo.write(str(restaurante) + "\n")

  def EscreverSaida(self, classificacao: Classificacao):
    arquivo = open(self.Endereco, "w")
    arquivo.write("Restaurantes classificados corretamente: %d" % (len(classificacao.ClassificadosCorretamente)) + "\n")

    for restaurante in classificacao.ClassificadosCorretamente:
      arquivo.write(restaurante.Nome + "\n")

    arquivo.write("\nRestaurantes classificados incorretamente: %d" % (len(classificacao.ClassificadosIncorretamente)) + "\n")

    for restaurante in classificacao.ClassificadosIncorretamente:
      arquivo.write(restaurante.Nome + "\n")

    arquivo.write("\nAcurácia do modelo: %.2f" % (classificacao.Acuracia) + "\n")
